domain/test/performance/CMakeLists.txt

```diff
@@ -1,42 +1,59 @@
 
-add_executable(hilbert_perf hilbert.cpp)
-target_include_directories(hilbert_perf PRIVATE ../../include)
-target_include_directories(hilbert_perf PRIVATE ../)
-target_link_libraries(hilbert_perf PRIVATE OpenMP::OpenMP_CXX)
-install(TARGETS hilbert_perf RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
+set(testname hilbert_perf)
+add_executable(${testname} hilbert.cpp)
+target_include_directories(${testname} PRIVATE ../../include)
+target_include_directories(${testname} PRIVATE ../)
+target_link_libraries(${testname} PRIVATE OpenMP::OpenMP_CXX)
+sphexa_add_test(${testname} EXECUTABLE ${testname} RANKS 1)
+install(TARGETS ${testname} RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
+unset(testname)
 
 if(CMAKE_CUDA_COMPILER)
-    add_executable(cudaNeighborsTest $<TARGET_OBJECTS:cuda_find_neighbors_obj> neighbor_driver.cu)
-    target_include_directories(cudaNeighborsTest PRIVATE ../../include)
-    target_include_directories(cudaNeighborsTest PRIVATE ../)
-    target_link_libraries(cudaNeighborsTest PRIVATE ${CUDA_RUNTIME_LIBRARY} OpenMP::OpenMP_CXX)
-    install(TARGETS cudaNeighborsTest RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
+    set(testname cudaNeighborsTest)
+    add_executable(${testname} $<TARGET_OBJECTS:cuda_find_neighbors_obj> neighbor_driver.cu)
+    target_include_directories(${testname} PRIVATE ../../include)
+    target_include_directories(${testname} PRIVATE ../)
+    target_link_libraries(${testname} PRIVATE ${CUDA_RUNTIME_LIBRARY} OpenMP::OpenMP_CXX)
+    sphexa_add_test(${testname} EXECUTABLE ${testname} RANKS 1)
+    install(TARGETS ${testname} RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
 
-    add_executable(octree_perf_gpu octree.cu)
-    target_include_directories(octree_perf_gpu PRIVATE ../../include)
-    target_include_directories(octree_perf_gpu PRIVATE ../)
-    install(TARGETS octree_perf_gpu RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
+    set(testname octree_perf_gpu)
+    add_executable(${testname} octree.cu)
+    target_include_directories(${testname} PRIVATE ../../include)
+    target_include_directories(${testname} PRIVATE ../)
+    sphexa_add_test(${testname} EXECUTABLE ${testname} RANKS 1)
+    install(TARGETS ${testname} RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
 
-    add_executable(hilbert_perf_gpu hilbert.cu)
-    target_include_directories(hilbert_perf_gpu PRIVATE ../../include)
-    target_include_directories(hilbert_perf_gpu PRIVATE ../)
-    install(TARGETS hilbert_perf_gpu RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
+    set(testname hilbert_perf_gpu)
+    add_executable(${testname} hilbert.cu)
+    target_include_directories(${testname} PRIVATE ../../include)
+    target_include_directories(${testname} PRIVATE ../)
+    sphexa_add_test(${testname} EXECUTABLE ${testname} RANKS 1)
+    install(TARGETS ${testname} RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR}/performance)
+    unset(testname)
 endif()
```

Comment:
All of these tests have identical includes, links and installs.
You could introduce a function `add_performance_test` to reduce the noise, like I did for the MPI integration tests.